### PR TITLE
chore: demo reupload with current master branch

### PR DESCRIPTION
**Pull request by ksiek127**
# Description
basic version of final demo
# Checklist

* [ ] Included code to test these changes
* [ ] Updated Jira

### CODE REVIEW

**Review by Monatyr — COMMENTED** [review]
Currently to link adjacent locations you have to use the following syntax (capital letters of cardinal directions matter!):

"directionToLocationMap": {
   "NORTH": "location-2",
   "SOUTH": "location-3"
  }

Other than that LGTM

File: configurations/demo-config-2/locations/location-0/objects/key-1.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "tag": "key-1",
+  "assetPath": "assets/key.png",
+  "position": { "row": 4, "col": 5 },
+  "onLeftClick": {"tag": "dialogue-action", "type": "dialogue", "statements": ["A key.,A, keyA, key.A, key.A, key.A, key.A, key.A, key.A key."], "assetPath": "assets/key.png"}
```
Comment (mhawryluk) [inline configurations/demo-config-2/locations/location-0/objects/key-1.json:5]: This dialogue doesn't make sense

File: configurations/demo-config-2/locations/location-0/objects/welcome-goblin.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "tag": "welcome-goblin",
+  "assetPath": "assets/goblin.png",
+  "onApproach": {"tag": "introduction", "type": "dialogue", "statements": ["Welcome!"]},
```
Comment (mhawryluk) [inline configurations/demo-config-2/locations/location-0/objects/welcome-goblin.json:4]: We could add a longer explanation of the aim of the game.

File: configurations/demo-config-2/locations/location-1/objects/chest-1.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+  "tag": "chest-1",
+  "onLeftClick": {"tag": "dialogue", "type": "dialogue", "statements": ["A very old chest."]},
```
Comment (mhawryluk) [inline configurations/demo-config-2/locations/location-1/objects/chest-1.json:3]: Remove or replace with an actual dialogue

File: configurations/demo-config-2/locations/location-1/objects/diamond-1.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{
+  "tag": "diamond-1",
+  "position": { "row": 7, "col": 2 },
+  "assetPath": "assets/diamond.png",
+  "onLeftClick": {
+    "tag": "get-item",
+    "type": "dialogue",
+    "statements": ["diamond"]
+  }
```
Comment (mhawryluk) [inline configurations/demo-config-2/locations/location-1/objects/diamond-1.json:9]: Replace with _collect_ action with assetPath and description

File: configurations/demo-config-2/locations/location-2/objects/goblin-1.json [hunk 5]
```diff
@@ -0,0 +1,12 @@
+{
+  "tag": "goblin-1",
+  "position": { "row": 4, "col": 5 },
+  "onLeftClick": {
+    "tag": "action",
+    "type": "dialogue",
+    "statements": [
+      "Far far away, behind the word mountains, far from the countries Vokalia and Consonantia, there live the blind texts. Separated they live in Bookmarksgrove right at the coast of the Semantics, a large language ocean. A small river named Duden flows by their place and supplies it with the necessary regelialia. It is a paradisematic country, in which roasted parts of sentences fly into your mouth. Even the all-powerful Pointing has no control about the blind texts it is an almost unorthographic life One day however a small line of blind text by the name of Lorem Ipsum decided to leave for the far World of Grammar. The Big Oxmox advised her not to do so, because there were thousands of bad Commas, wild Question Marks and devious Semikoli, but the Little Blind Text didn’t listen."
```
Comment (mhawryluk) [inline configurations/demo-config-2/locations/location-2/objects/goblin-1.json:8]: This text was just a placeholder for an actual dialogue

File: configurations/demo-config-2/locations/location-4/objects/chest-1.json [hunk 6]
```diff
@@ -0,0 +1,6 @@
+{
+  "tag": "chest-1",
+  "position": { "row": 4, "col": 3 },
+  "onLeftClick": {"tag": "chest", "type": "dialogue", "statements": ["another chest"]},
```
Comment (mhawryluk) [inline configurations/demo-config-2/locations/location-4/objects/chest-1.json:4]: !!

File: configurations/demo-config-2/locations/location-5/objects/diamond-1.json [hunk 7]
```diff
@@ -0,0 +1,10 @@
+{
+  "tag": "diamond-1",
+  "position": { "row": 8, "col": 9 },
+  "assetPath": "assets/diamond.png",
+  "onLeftClick": {
+    "tag": "get-item",
+    "type": "dialogue",
+    "statements": ["diamond"]
+  }
```
Comment (mhawryluk) [inline configurations/demo-config-2/locations/location-5/objects/diamond-1.json:9]: _collect_ action instead

**Review by mhawryluk — CHANGES_REQUESTED** [review]

**Review by kkafar — DISMISSED** [review]
Some bullet points from me:

* We need to take advantage of all available mechanics (especially those new: conditional actions, before & after actions, battle.
* We need a story here. The game must feel engaging & the player must know what its purpose is.

**Review by mhawryluk — APPROVED** [review]
Fantasticc

**Pull request merged**